spec/models/like_spec.rb

```diff
@@ -0,0 +1,18 @@
+require 'rails_helper'
+
+RSpec.describe Like, type: :model do
+  user = User.new(name: 'Klaus', photo: 'Mikelson.png', bio: 'Always and forever', post_counter: 0)
+  user.save
+  post = Post.new(author_id: user, title: 'Hello Elijah', text: 'How is mum')
+  post.save
+
+  user.likes.create(post_id: post, author_id: user)
+
+  it 'update likes counter' do
+    expect(user.likes.length).to eq 1
+  end
+
+  it 'update post like counter' do
+    expect(subject.post.likes_counter).to eq(1)
+  end
+end
```

Comment:
- [ ] Your tests are failing to run in the first place due to some error. We will use this `like_spec.rb` to dissect the issue.

  - The issue of `user_id` not being related to like is because in the User model you did not define the foreign key as `author_id`, so Rails is looking for a `user_id`. You need to modify the `has_many` in the User model to account for the foreign key.

    ```
    has_many :likes, foreign_key: :author_id
    ```

    ![image](https://user-images.githubusercontent.com/36542986/235616777-24a4fdb0-660a-4788-9505-05f1b35d4382.png)

  - After resolving that the next issue that comes up is related to the test on line 16. To begin with, you need to remove `subject` from line 16 as there is nowhere you defined it.

    ```
    expect(post.likes_counter).to eq(1)
    ```
  
    From the error message, you are trying to check for `likes_counter` for an item that does not even exist in the first place (nil:NilClass). The post creation on line 6 does not really work because of failed validations. You need to initialize `comments_counter` and `likes_counter` attributes. Also your `author_id` should be `user.id`, not just `user`

    ```
    post = Post.new(author_id: user.id, title: 'Hello Elijah', text: 'How is mum', comments_counter: 0, likes_counter: 0)
    ```

    ![image](https://user-images.githubusercontent.com/36542986/235617090-f72b4472-3f3f-44df-9efe-4cf025aa2f85.png)

  - Next your test on line 16 would not still pass because the `update_posts_likes_counter` method in your Like model is not being called at all.

    Instead of creating your like with this code: `user.likes.create(post_id: post, author_id: user)`, you may need to create your like in this manner:

    ```
    like = Like.create(post:, author: user)
    ````
  
    You need to call an ActiveRecord `after_save` callback to the Like model so that whenever a like is saved it automatically calls the method. This is what your Like model should look like:

    ```
    class Like < ApplicationRecord
      belongs_to :author, class_name: 'User'
      belongs_to :post
      after_save :update_posts_likes_counter
    
      private
    
      def update_posts_likes_counter
        post.increment!(:likes_counter)
      end
    end
    ```

    ![okdue](https://user-images.githubusercontent.com/36542986/235621801-c58727ae-a2f9-4181-b483-d8dbd42bdd58.jpg)